.github/workflows/main.yml

```diff
@@ -0,0 +1,38 @@
+name: Linting and Unit Tests
+
+# only run this workflow on new commits to main
+# or PRs into main
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+  schedule:
+    # Run every Monday morning at 11:00a UTC, 6:00a CST
+    - cron: '0 11 * * 1'
```

Comment:
Why do we need it to run on a schedule?

Reply:
running CI on a schedule will catch any issues with anything that is unpinned. That can prevent surprises where something else breaks at the time when you submit a PR. So it just smooths out the pain of adjusting to new releases of dependencies breaking things.

Reply:
This is especially nice for code that doesn't change very often.